### PR TITLE
feat: Added UpgradeHandler to burn unclaimed airdrop tokens

### PR DESCRIPTION
**Pull request by vishal-kanna**
This PR consists of UpgradeHandler which burns the tokens of a specified address

### CODE REVIEW

File: app/upgrades/v2.3.1/upgrade.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	auth "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	distribution "github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/envadiv/Passage3D/app/upgrades"
+	claim "github.com/envadiv/Passage3D/x/claim/keeper"
+)
+
+const Name = "v2.3.1"
```
Comment (anilcse) [inline app/upgrades/v2.3.1/upgrade.go:16]:
```suggestion
const Name = "v2.4.0"
```

File: app/upgrades/v2.3.1/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			return nil, err
+		}
+
+		return fromVM, nil
+	}
+}
+
+func ExecuteProposal(ctx sdk.Context, ak auth.AccountKeeper, bk bank.Keeper) error {
+	// get account address
+	addr, err := sdk.AccAddressFromBech32("pasg1emkh9v2kk03j4ccs0pnzk78e7ejq6wlz8mnn9u")
```
Comment (anilcse) [inline app/upgrades/v2.3.1/upgrade.go:44]:
```suggestion
        // airdrop claim module account
	addr, err := sdk.AccAddressFromBech32("pasg1m5dncvfv7lvpvycr23zja93fecun2kcvpfszyd")
```

File: app/upgrades/v2.3.1/upgrade.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	if err != nil {
+		return err
+	}
+	// get the balances from the account
```
Comment (anilcse) [inline app/upgrades/v2.3.1/upgrade.go:48]:
```suggestion

	// get the balances from the account
```

File: app/upgrades/v2.3.1/upgrade.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	balances := bk.GetAllBalances(ctx, addr)
+	// send the tokens to the module account
```
Comment (anilcse) [inline app/upgrades/v2.3.1/upgrade.go:50]:
```suggestion

	// send the tokens to the module account
```

File: app/upgrades/v2.3.1/upgrade.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	if err := bk.SendCoinsFromAccountToModule(ctx, addr, govtypes.ModuleName, balances); err != nil {
+		return err
+	}
+	// burn the coins
```
Comment (anilcse) [inline app/upgrades/v2.3.1/upgrade.go:54]:
```suggestion

	// burn the coins
```

File: app/upgrades/v2.3.1/upgrade.go (hunk 4 above)
Comment (anilcse) [reply app/upgrades/v2.3.1/upgrade.go:50]: Why this is required? Can't we burn the tokens from account directly?

**Review by anilcse — CHANGES_REQUESTED** [review]

File: app/upgrades/v2.3.1/upgrade.go (hunk 4 above)
Comment (aleem1314) [reply app/upgrades/v2.3.1/upgrade.go:50]: The claim module account does not have permission to burn tokens. We're moving the funds to another module account to burn them instead.

File: app/upgrades/v2.3.1/upgrade.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+func ExecuteProposal(ctx sdk.Context, ak auth.AccountKeeper, bk bank.Keeper) error {
+	// get account address
+	addr, err := sdk.AccAddressFromBech32("pasg1emkh9v2kk03j4ccs0pnzk78e7ejq6wlz8mnn9u")
+	if err != nil {
+		return err
+	}
+	// get the balances from the account
+	balances := bk.GetAllBalances(ctx, addr)
+	// send the tokens to the module account
+	if err := bk.SendCoinsFromAccountToModule(ctx, addr, govtypes.ModuleName, balances); err != nil {
```
Comment (akhilkumarpilli) [inline app/upgrades/v2.3.1/upgrade.go:51]:
```suggestion
	if err := bk.SendCoinsFromModuleToModule(ctx, claimtypes.ModuleName, govtypes.ModuleName, balances); err != nil {
```

File: app/upgrades/v2.4.0/upgrade.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+func ExecuteProposal(ctx sdk.Context, ak auth.AccountKeeper, bk bank.Keeper) error {
+	// get airdrop claim module account
+	addr := ak.GetModuleAddress(claimtypes.ModuleName)
+
+	// get the balances from the airdrop claim module account
+	balances := bk.GetAllBalances(ctx, addr)
+
+	// claim moudle account do not have the burner permissions
+	// so we are sending it to the another module account and burning the tokens
```
Comment (akhilkumarpilli) [inline app/upgrades/v2.4.0/upgrade.go:51]:
```suggestion
	// get all balances of the airdrop claim module account
	balances := bk.GetAllBalances(ctx, addr)

	// claim module account do not have the burner permissions
	// so we are sending it to the another module account and burning the tokens
```

**Review by anilcse — APPROVED** [review]

**Pull request merged**